packages/record-data/addon/-private/record-data.ts

```diff
@@ -98,7 +99,7 @@ export default class RecordDataDefault implements RelationshipRecordData {
       changedKeys = this._changedKeys(data.attributes);
     }
 
-    assign(this._data, data.attributes);
+    assign(this._data, data.attributes || {});
```

Comment:
Hmm, this seems odd. `Ember.assign` (and Object.assign) both allow for passed arguments to be `null` or `undefined` and skip them. I think this fallback should be removed.

```suggestion
    assign(this._data, data.attributes);
```

Reply:
The types for Ember.assign say otherwise :'(

Reply:
We tried to overload but I think they weren't getting discovered by vscode, I think now that I know the mechanism by which vscode trolls us (basically never trust vscode) likely the overload was working and we can revert this change.